openpgp/keys.go

```diff
@@ -255,7 +255,10 @@ func (el EntityList) KeysByIdUsage(id uint64, requiredUsage byte) (keys []Key) {
 				usage |= packet.KeyFlagEncryptStorage
 			} else if key.PublicKey.PubKeyAlgo == packet.PubKeyAlgoDSA {
 				usage |= packet.KeyFlagSign
+			} else if key.Entity.PrimaryKey.KeyId == id {
+				usage = (packet.KeyFlagCertify | packet.KeyFlagSign | packet.KeyFlagEncryptCommunications | packet.KeyFlagEncryptStorage)
```

Comment:
Would another possible fix be to add an RSA-specific block here, similar to what we have above with DSA and ElGamal? (Or maybe, one for each of the 3 different usage-specific RSA aliases that rfc4480 defines for G-d only knows what reason?) I'm a little worried about this branch deciding that a key is usable for encryption when it really isn't.